src/core/cstring.h

```diff
@@ -34,7 +34,7 @@ class CString
     explicit CString(const CString& cstr);
     ~CString();
 
-    uint8_t* data();
+    uint8_t* data() const;
```

Comment:
If the getter is const, then the returned pointer needs to be const too.
Otherwise if the caller can modify the string data, the function can't be const!